cmd/rpc.go

```diff
@@ -165,6 +165,15 @@ type RPCClient struct {
 }
 
 func (client *RPCClient) setRetryTicker(ticker *time.Ticker) {
+	if ticker == nil {
```

Comment:
unable to understand why this `if` block is needed.

Reply:
The reason is the entire operation of setRetryTicker blocks under an on-going RPC operation

Reply:
Holding a read lock for entire RPC operation to complete is not acceptable since there are other pending calls in transit which are in the state of resetting the ticker which will hold write locks, which end up block on each other. 

Check the profiling output here

[01.pdf](https://github.com/minio/minio/files/2257758/01.pdf)

Reply:
Latest changes made things clear here